eth/tracers/api.go

```diff
@@ -892,7 +895,11 @@ func (api *API) TraceCall(ctx context.Context, args ethapi.TransactionArgs, bloc
 	if config != nil && config.Reexec != nil {
 		reexec = *config.Reexec
 	}
-	statedb, release, err := api.backend.StateAtBlock(ctx, block, reexec, nil, true, false)
+	if config != nil && config.TxIndex != nil {
```

Comment:
Please add a comment for the method how exactly state is computed based on these parameters. I.e. "latest" without any transaction index means we're simulating on the state after the latest block has been executed. Whereas "latest" with a transaction index means we're simulating on the intermediate state after a given tx in the "latest" block. It might be a bit unintuitive that "txIdx" sort of rolls back the state.